jdbc/src/main/java/software/amazon/timestream/jdbc/TimestreamDriver.java

```diff
@@ -175,9 +176,47 @@ public boolean jdbcCompliant() {
      * @return the name of the currently running application.
      */
     private static String getApplicationName() {
-        // Currently not supported.
-        // Need to implement logic to get the process ID of the current process, then check the set of running processes and pick out
+        // What we do is get the process ID of the current process, then check the set of running processes and pick out
         // the one that matches the current process. From there we can grab the name of what is running the process.
+        try {
+            final String pid = ManagementFactory.getRuntimeMXBean().getName().split("@")[0];
+            final boolean isWindows = System.getProperty("os.name").startsWith("Windows");
+
+            String command;
+            if (isWindows) {
+                command = "tasklist /fo csv /nh";
+            } else {
+                command = "ps -eo pid,comm";
+            }
+
+            final Process process = Runtime.getRuntime().exec(command);
+            try (BufferedReader input = new BufferedReader(
+                    new InputStreamReader(process.getInputStream(), StandardCharsets.UTF_8))) {
+                String line;
+                if (isWindows) {
+                    while ((line = input.readLine()) != null) {
+                        int start = line.indexOf(",");
+                        int end = line.indexOf(",", start+1);
+                        if (line.substring(start+1, end).contains(pid)){
+                            return line.substring(0, line.indexOf(","));
+                        }
+                    }
+                } else {
+                    while ((line = input.readLine()) != null) {
+                        line = line.trim();
+                        if (line.startsWith(pid)) {
+                            return line.substring(line.indexOf(" ") + 1);
```

Comment:
On macOS, this can lead to including path. The path may contain sensitive information like an OS username. This data can not be collected and must be excluded; only application names can be collected. 

@RoyZhang2022, this logic has to be fixed, so the username is not being collected.

Reply:
It is fixed.